Restore-TeamsEV.ps1

```diff
@@ -1,133 +1,173 @@
-	If ($OverrideAdminDomain) {
+if ((Get-PSSession | Where-Object -FilterScript {
+			$_.ComputerName -like '*.online.lync.com'
+		}).State -eq 'Opened')
+{
+	Write-Host -Object 'Using existing session credentials'
+}
+else
+{
+	Write-Host -Object 'Logging into Office 365...'
+	if ($OverrideAdminDomain)
+	{
 		$O365Session = New-CsOnlineSession -OverrideAdminDomain $OverrideAdminDomain
 	}
-	Else {
+	else
+	{
 		$O365Session = New-CsOnlineSession
 	}
-	Import-PSSession $O365Session -AllowClobber
+	Import-PSSession -Session $O365Session -AllowClobber
 }
 
-$EV_Entities = "Dialplans", "VoiceRoutes", "VoiceRoutingPolicies", "PSTNUsages", "TranslationRules", "PSTNGateways"
+$EV_Entities = 'Dialplans', 'VoiceRoutes', 'VoiceRoutingPolicies', 'PSTNUsages', 'TranslationRules', 'PSTNGateways'
 
-Write-Host "Validating backup files." 
+Write-Host -Object 'Validating backup files.'
 
-ForEach ($EV_Entity in $EV_Entities) {
-	Try {
+foreach ($EV_Entity in $EV_Entities)
+{
+	try
+	{
 		$ZipItem = $ZipStream.GetEntry("$EV_Entity.txt")
-		$ItemReader = New-Object System.IO.StreamReader($ZipItem.Open())
-
+		$ItemReader = New-Object -TypeName System.IO.StreamReader -ArgumentList ($ZipItem.Open())
+		
 		Set-Variable -Name $EV_Entity -Value ($ItemReader.ReadToEnd() | ConvertFrom-Json)
-		If ((Get-Variable $EV_Entity).Value[0].Identity -eq $NULL) {Throw("Error")} # Throw error if there is no Identity field, which indicates this isn't a proper backup file
+		
+		if ((Get-Variable -Name $EV_Entity).Value[0].Identity -eq $NULL)
+		{
+			throw ('Error')
+		} # Throw error if there is no Identity field, which indicates this isn't a proper backup file
 	}
-	Catch {
-		Write-Error "$EV_Entity.txt could not be found or could not be parsed. Exiting."
-		Exit
+	catch
+	{
+		Write-Error -Message "$EV_Entity.txt could not be found or could not be parsed. Exiting."
+		exit
 	}
 }
 
-Write-Host -ForegroundColor Green "Backup files are OK!"
+Write-Host -ForegroundColor Green -Object 'Backup files are OK!'
 
-If (!$KeepExisting) {
-	$Confirm = Read-Host "WARNING: This will ERASE all existing dialplans/voice routes/policies etc prior to restoring from backup. Continue (Y/N)?"
-	If ($Confirm -notmatch "^[Yy]$") {
-		Write-Host "Exiting without making changes."
-		Exit
+if (!$KeepExisting)
+{
+	$Confirm = Read-Host -Prompt 'WARNING: This will ERASE all existing dialplans/voice routes/policies etc prior to restoring from backup. Continue (Y/N)?'
+	if ($Confirm -notmatch '^[Yy]$')
+	{
+		Write-Host -Object 'Exiting without making changes.'
+		exit
 	}
 	
-	Write-Host "Erasing all existing dialplans/voice routes/policies etc."
+	Write-Host -Object 'Erasing all existing dialplans/voice routes/policies etc.'
+	
 	Get-CsTenantDialPlan | Remove-CsTenantDialPlan
 	Get-CsOnlineVoiceRoute | Remove-CsOnlineVoiceRoute
 	Get-CsOnlineVoiceRoutingPolicy | Remove-CsOnlineVoiceRoutingPolicy
-	Set-CsOnlinePstnUsage Global -Usage $NULL
+	Set-CsOnlinePstnUsage -ErrorAction Global -Usage $NULL

```

Comment:
I need to fix this, sorry

Reply:
Corrected:
```powershell
Set-CsOnlinePstnUsage -Identity Global -Usage $NULL -ErrorAction SilentlyContinue
```

Reply:
I gotta figure out how to use Github now

Reply:
Should I do a new one?

Reply:
Yeah, if you could. I'm very new at this Github stuff, so be patient. Most of your changes are definitely for the better, and I appreciate it very much. I'm just a stickler for my personal style, which includes the capitalization and leading brace on the same line, rather than on the line below. 

Curiously enough, I was chatting with a buddy of mine (Pat Richard) and discussing how to use Github properly. He was showing me some of his stuff on Github via screenshare, and you were making changes to his stuff as well. You must be very active!

Reply:
No rush @kenlasko :)
If I find something interesting, I sometimes work a bit on it. Like it did with your scripts.
Different story with the script from Pat: I was working on exactly the same when I found his script. So I dropped mine and adopted his approach.
And as huge fan of open source and Powershell, I always think it might be a good idea to submit my reworked stuff back.

I will close this request and submit a new one with the fixed line tomorrow. If you like it, just merge it... if not, just close it without merging it back. That’s the fun part of it: you can decide.

I will do a restore test in a new Tenant tomorrow, just to see what happens. I did a backup with a customized configuration, from my production Tenant.